code/datums/components/rotation.dm

```diff
@@ -129,41 +119,59 @@
 			rot_degree = 90
 		if(ROTATION_FLIP)
 			rot_degree = 180
-	AM.setDir(turn(AM.dir,rot_degree))
-	after_rotation.Invoke(user,rotation_type)
+	rotated_obj.setDir(turn(rotated_obj.dir, rot_degree))
+	after_rotation.Invoke(user, rotation_type)
 
 /datum/component/simple_rotation/proc/default_can_user_rotate(mob/living/user, rotation_type)
-	if(!istype(user) || !user.canUseTopic(parent, BE_CLOSE, NO_DEXTERITY, FALSE, !iscyborg(user)))
+	if(istype(user) && user.canUseTopic(parent, BE_CLOSE, NO_DEXTERITY, FALSE, !iscyborg(user)))
+		return TRUE
+	if(isobserver(user) && CONFIG_GET(flag/ghost_interaction) && (src.rotation_flags & ROTATION_GHOSTS_ALLOWED))
+		return TRUE	
+	return FALSE
+
+/datum/component/simple_rotation/proc/default_can_be_rotated(mob/living/user, rotation_type)
+	var/obj/rotated_obj = parent
+
+	if(src.rotation_flags & ROTATION_REQUIRE_WRENCH)
+		if(!istype(user))
+			return FALSE
+		var/obj/item/tool = user.get_active_held_item()
+		if(!tool || tool.tool_behaviour != TOOL_WRENCH)
+			rotated_obj.balloon_alert(user, "need a wrench")
+			return FALSE
+	if(src.rotation_flags & ROTATION_IGNORE_ANCHORED) // used to ignore chairs being anchored
+		return TRUE
+	if(rotated_obj.anchored)
+		rotated_obj.balloon_alert(user, "need to unwrench")
```

Comment:
```suggestion
	if(!(rotation_flags & ROTATION_IGNORE_ANCHORED) && rotated_obj.anchored)
		rotated_obj.balloon_alert(user, "need to unwrench")
```
Seems like this one is linked to the other check? Seems odd to decouple it, but feel free to ignore the suggestion if you think otherwise.